### ISSUE TITLE
Newlines cause browser-sync js snippet to not be inserted when using Apache SSI, bs proxy mode

### ISSUE THREAD

**Issue by nvioli** [BrowserSync/browser-sync#525]
I have some html files using SSI to include the `<body>` tag, and browser-sync fails to insert the js snippet on these pages. I'm running bs from gulp, using the proxy option, with pretty barebones config:

    gulp.task('browser-sync', function() {
       browserSync({
          proxy: localhost:8080
       });
    });

On the source side, I'm able to reproduce the issue with just an SSI call in the target html document:

      

where body.html just has

    <body></body>

When the include is the first (only) line in the html file, the script is injected properly, however when there's a bare newline before it, it's not inserted and hence bs does not connect. Clearly the html returned from Apache should be identical.

Please let me know if I can provide any more information.

**Comment by shakyShane** [comment]
A file with this content works for me

```
\n<body></body>
```

?

**Comment by nvioli** [comment]
Yes, that works ok for me too; the problem somehow only manifests when using the SSI call, which is totally baffling.

**Comment by balazsgabor** [comment]
I have similar issue with SSI & browser-sync. The script is not injected, but if I save the output of the site to a static HTML file, and try to fetch that, the script will be injected without having any modifications to the file.

I did some further investigation, and found 2 problems with resp-modifier:

1. The `isHtml()` function does not handle the UTF-8 BOM charachter (EF BB BF), only UTF-16 (FE FF or FF FE)
2. If the pattern we would like to match, comes in separate chunks, the regex will never match. For example in my case these were the chunks, with which the `res.write` was called:

```

</head>
<body 
```
```  
class=" 
```
``` 
downloads software-downloads" 
```
```
>
    <div i...
```

The `exists()` function will be called on each chunk separately. This way the default `/<body[^>]*>/i` pattern can never match.

**Comment by iam4x** [comment]
Anyone have found a workaround? I have to include `browser-sync-xxx.js` manually and that's sucks!

**Comment by shakyShane** [comment]
@balazsgabor your second point is fixed in latest releases.
@nvioli can you also try the following and see if this is still a problem.

```sh
npm rm browser-sync && npm i browser-sync@latest -D
```

... to get latest dependency updates

**Comment by nvioli** [comment]
@shakyShane thanks for the effort but I still have the same problem, reproducible as in the original post.

Please let me know if I can provide any more information.

**Comment by Amberlamps** [comment]
@nvioli Thanks for this posting. I am using dustjs as template engine and I have this little line in my server code:

```
dust.config.whitespace = app.get('env') !== 'development';
```

Remove all whitespaces when not in `development` mode. However, when starting app in `development` mode and passing url to browserSync as proxy, browserSync was not injecting its snippet anymore.

I do not have a solution for this, but I think this is a bug. I am setting `dust.config.whitespace = false` in any environment mode to get browserSync running, but I actually want whitespaces to be rendered when in `development` mode for better debugging.
